src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -104,6 +117,23 @@ public CCDCaseApi(
         return claims.isEmpty() ? Optional.empty() : Optional.of(claims.get(0));
     }
 
+    public Optional<Long> getOnHoldIdByExternalId(String externalId, String authorisation) {
+        User user = userService.getUser(authorisation);
+        List<CaseDetails> result = searchAll(user, ImmutableMap.of("case.externalId", externalId));
```

Comment:
why don't you search on hold cases instead of search all? then you can remove that other exception case below?